### PR TITLE
fix: mWeb - Selector - User lands on the same WS when double tapping on different one on selector.

### PR DESCRIPTION
**Pull request by Krishna2323**
### Explanation of Change
- This PR uses `isFocused` state to prevent multiple clicks on an option in workspace switcher page which was leading to multiple navigation issue.

### Fixed Issues
$ https://github.com/Expensify/App/issues/51402
PROPOSAL: https://github.com/Expensify/App/issues/51402#issuecomment-2435323594

### Tests
1. Tap on Workspace selector on the top left corner.
2. Change to any workspace on the list.
3. Double tap on Expensify option.
4. Verify that all the chats are displayed in LHN and `Expensify` is selected.
5. Open Workspace selector again
6. Double tap on the next workspace to the one selected.
7. Verify that the chats of the selected workspace are displayed and the selected workspace is shown as the icon of workspace switcher selector.

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Tap on Workspace selector on the top left corner.
2. Change to any workspace on the list.
3. Double tap on Expensify option.
4. Verify that all the chats are displayed in LHN and `Expensify` is selected.
5. Open Workspace selector again
6. Double tap on the next workspace to the one selected.
7. Verify that the chats of the selected workspace are displayed and the selected workspace is shown as the icon of workspace switcher selector.

### QA Steps
1. Tap on Workspace selector on the top left corner.
2. Change to any workspace on the list.
3. Double tap on Expensify option.
4. Verify that all the chats are displayed in LHN and `Expensify` is selected.
5. Open Workspace selector again
6. Double tap on the next workspace to the one selected.
7. Verify that the chats of the selected workspace are displayed and the selected workspace is shown as the icon of workspace switcher selector.

// TODO: These must be filled out, or the issue title must include "[No QA]."

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/8858a224-f46b-44b0-b7c6-2ee62c345554

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/d89cbb97-59c6-4b43-a260-3c0e8c7a5ba0

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/93761a86-ecab-442e-b286-9d6e77f0a1b6

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/b27fdc7c-83f2-46c4-bbfe-c5f3b5572861

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/94d4361c-6080-43f5-8143-d24648298fb3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/059b068d-7930-4f37-b790-9f7c38faa118

</details>

### CODE REVIEW

**Comment by s77rt** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/user-attachments/assets/3eb331a1-258a-4a63-aa19-836af2adea03

</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/user-attachments/assets/9b60f105-4959-4d91-aa36-967432d662b2

</details>

<details>
<summary>iOS: Native</summary>


https://github.com/user-attachments/assets/25c500c5-5607-4680-a243-9e1f4bff0333

</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/user-attachments/assets/314a895b-0044-4ad3-9dc0-ac3afbf4aedc

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/user-attachments/assets/da822a7a-7eae-4b45-8462-8c9f1dc26573

</details>

<details>
<summary>MacOS: Desktop</summary>









https://github.com/user-attachments/assets/3311c7de-601e-4a9b-83ba-80279af34a64






</details>

**Comment by s77rt** [comment]
The test unit should fail if we remove the `!isFocused` from `WorkspaceSwitcherPage` but currently it does not because the solution is implemented in the test itself

**Comment by s77rt** [comment]
I don't see how the latest change will help. You should render the App (and the Workspace Switcher Page) from the app source and not by defining a function in the test. The function in the test will always pass since it will always has the solution in place.

**Comment by s77rt** [comment]
Not a problem! Please use [PaginationTest](https://github.com/Expensify/App/blob/7bfb34527508b47732cab6d4280a943ffab48066/tests/ui/PaginationTest.tsx) as a reference instead. We render the `<App />` and add onyx sample data (workspaces in this case) then we navigate to the workspace switcher and fire the press button twice.

File: tests/ui/WorkspaceSwitcherPageTest.tsx
Comment (s77rt) [inline tests/ui/WorkspaceSwitcherPageTest.tsx:1]: Remove this file

File: tests/ui/WorkspaceSwitcherTest.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+jest.mock('@libs/Permissions');
+jest.mock('@components/Icon/Expensicons');
+jest.mock('@src/hooks/useActiveWorkspaceFromNavigationState');
+jest.mock('@src/hooks/useResponsiveLayout');
+
+TestHelper.setupApp();
+
+async function navigateToWorkspaceSwitcher(): Promise<void> {
+    const hintText = Localize.translateLocal('workspace.switcher.headerTitle');
+    const optionRow = await screen.findByTestId(hintText);
```
[the remark this answers is not in the document]
Comment (s77rt) [reply tests/ui/WorkspaceSwitcherTest.tsx:33]: The `MockedSidebarLinks` is missing the `<TopBar />`

**Comment by s77rt** [comment]
@Krishna2323 Can you please let me know when this is ready for review

**Comment by s77rt** [comment]
```tsx
import * as NativeNavigation from '@react-navigation/native';
import {act, fireEvent, render, screen} from '@testing-library/react-native';
import Onyx from 'react-native-onyx';
import * as Localize from '@libs/Localize';
import * as AppActions from '@userActions/App';
import * as User from '@userActions/User';
import App from '@src/App';
import ONYXKEYS, {OnyxKey} from '@src/ONYXKEYS';
import type {NativeNavigationMock} from '../../__mocks__/@react-navigation/native';
import * as LHNTestUtils from '../utils/LHNTestUtils';
import PusherHelper from '../utils/PusherHelper';
import * as TestHelper from '../utils/TestHelper';
import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
import waitForBatchedUpdatesWithAct from '../utils/waitForBatchedUpdatesWithAct';

// We need a large timeout here as we are lazy loading React Navigation screens and this test is running against the entire mounted App
jest.setTimeout(60000);

jest.mock('@react-navigation/native');

TestHelper.setupApp();

async function navigateToWorkspaceSwitcher(): Promise<void> {
    const workspaceSwitcherButton = await screen.findByTestId('WorkspaceSwitcherButton');
    fireEvent(workspaceSwitcherButton, 'press');
    await act(() => {
        (NativeNavigation as NativeNavigationMock).triggerTransitionEnd();
    });
    await waitForBatchedUpdatesWithAct();
}

async function signInAndGetApp(): Promise<void> {
    // Render the App and sign in as a test user.
    render(<App />);
    await waitForBatchedUpdatesWithAct();

    const hintText = Localize.translateLocal('loginForm.loginForm');
    const loginForm = await screen.findAllByLabelText(hintText);
    expect(loginForm).toHaveLength(1);

    await act(async () => {
        await TestHelper.signInWithTestUser();
    });
    await waitForBatchedUpdatesWithAct();

    User.subscribeToUserEvents();
    await waitForBatchedUpdates();

    AppActions.setSidebarLoaded();

    await waitForBatchedUpdatesWithAct();
}

describe('WorkspaceSwitcherPage', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        Onyx.clear();

        // Unsubscribe to pusher channels
        PusherHelper.teardown();
    });

    it('triggers press on workspaces list item only once', async () => {
        await signInAndGetApp();

        await Onyx.mergeCollection(ONYXKEYS.COLLECTION.POLICY, {
            [`${ONYXKEYS.COLLECTION.POLICY}1` as const]: LHNTestUtils.getFakePolicy('1', 'Workspace A'),
            [`${ONYXKEYS.COLLECTION.POLICY}2` as const]: LHNTestUtils.getFakePolicy('2', 'Workspace B'),
            [`${ONYXKEYS.COLLECTION.POLICY}3` as const]: LHNTestUtils.getFakePolicy('3', 'Workspace C'),
        });

        await navigateToWorkspaceSwitcher();

        const workspaceRow = screen.getByLabelText('Workspace A');
        expect(workspaceRow).toBeOnTheScreen();
    });
});
```

Here a simplified test that you can work on top of. Basically now we need to press on Workspace A once and verify it's selected. Then press on workspace B twice and verify that B is selected.


Let me know if you have any questions

**Comment by s77rt** [comment]
@Krishna2323 Wrap the calls in `act`

**Comment by s77rt** [comment]
@Krishna2323 Instead of trying to fire the press event twice, mock `useIsFocused` to return `false` and fire the press event and verify that we weren't navigated. Then make the mock return true and this time we should get navigated

File: src/types/onyx/Policy.ts
Comment (s77rt) [inline src/types/onyx/Policy.ts:1]: Revert

File: tests/ui/WorkspaceSwitcherTest.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+jest.setTimeout(60000);
+
+jest.mock('@react-navigation/native', () => {
+    const actualNav = jest.requireActual<typeof Navigation>('@react-navigation/native');
+    return {
+        ...actualNav,
+        useIsFocused: jest.fn(),
+        triggerTransitionEnd: jest.fn(),
+    };
+});
```
Comment (s77rt) [inline tests/ui/WorkspaceSwitcherTest.tsx:28]: Can we not use the `actualNav` and just return `useIsFocused`

File: tests/ui/WorkspaceSwitcherTest.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+TestHelper.setupApp();
+
+async function navigateToWorkspaceSwitcher(): Promise<void> {
+    const workspaceSwitcherButton = await screen.findByTestId('WorkspaceSwitcherButton');
+    fireEvent(workspaceSwitcherButton, 'press');
+    await act(() => {
+        (NativeNavigation as NativeNavigationMock).triggerTransitionEnd();
+    });
+    await waitForBatchedUpdatesWithAct();
+}
```
Comment (s77rt) [inline tests/ui/WorkspaceSwitcherTest.tsx:38]: NAB. Move this below `signInAndGetApp`

File: tests/ui/WorkspaceSwitcherTest.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+        await Onyx.mergeCollection(ONYXKEYS.COLLECTION.POLICY, {
+            [`${ONYXKEYS.COLLECTION.POLICY}1` as const]: LHNTestUtils.getFakePolicy('1', 'Workspace A'),
+            [`${ONYXKEYS.COLLECTION.POLICY}2` as const]: LHNTestUtils.getFakePolicy('2', 'Workspace B'),
+            [`${ONYXKEYS.COLLECTION.POLICY}3` as const]: LHNTestUtils.getFakePolicy('3', 'Workspace C'),
+        });
+
+        await navigateToWorkspaceSwitcher();
+
+        // When `isFocused` is true, pressing on a workpace should navigate and the option should be removed from the screen
```
Comment (s77rt) [inline tests/ui/WorkspaceSwitcherTest.tsx:83]:
```suggestion
```
Unnecessary comment

File: tests/ui/WorkspaceSwitcherTest.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        const workspaceRowB = screen.getByLabelText('Workspace B');
+        fireEvent.press(workspaceRowB);
+        expect(screen.queryByLabelText('Workspace B')).toBeNull();
```
Comment (s77rt) [inline tests/ui/WorkspaceSwitcherTest.tsx:86]: Use the `WorkspaceSwitcherPage` test id instead

**Review by s77rt — CHANGES_REQUESTED** [review]

File: tests/ui/WorkspaceSwitcherTest.tsx (hunk 4 above)
Comment (s77rt) [reply tests/ui/WorkspaceSwitcherTest.tsx]: Please update for both tests

File: tests/ui/WorkspaceSwitcherTest.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+            [`${ONYXKEYS.COLLECTION.POLICY}1` as const]: LHNTestUtils.getFakePolicy('1', 'Workspace A'),
+            [`${ONYXKEYS.COLLECTION.POLICY}2` as const]: LHNTestUtils.getFakePolicy('2', 'Workspace B'),
+            [`${ONYXKEYS.COLLECTION.POLICY}3` as const]: LHNTestUtils.getFakePolicy('3', 'Workspace C'),
+        });
+
+        await navigateToWorkspaceSwitcher();
+
+        const workspaceRowB = screen.getByLabelText('Workspace B');
+        fireEvent.press(workspaceRowB);
+        expect(screen.queryByLabelText('Workspace B')).toBeNull();
```
Comment (s77rt) [inline tests/ui/WorkspaceSwitcherTest.tsx:85]: This is not updated yet

**Review by s77rt — APPROVED** [review]

**Review by blimpich — APPROVED** [review]
Great work! Thank you for writing the test! Not easy to initialize a test suite for this case but our codebase is better for it!

**Pull request merged**